sound/soc/sof/pcm.c

```diff
@@ -74,14 +74,19 @@ static int sof_pcm_hw_params(struct snd_pcm_substream *substream,
 		dev_err(sdev->dev, "error: could not allocate %d bytes for PCM %d\n",
 			params_buffer_bytes(params), spcm->pcm.pcm_id);
 		return ret;
+	} else if (ret == 1) {
```

Comment:
@libinyang good catch. But it would be more readable if you can just do if(ret) instead of else if here

Reply:
@ranj063 
The snd_pcm_lib_malloc_pages returns:
 "Return: 1 if the buffer is changed, 0 if not changed, or a negative code on failure."

This is why I use "ret == 1".

Reply:
@libinyang its the same as saying if (ret) instead of elseif (ret == 1)

Reply:
I got it. I will update.